### PR TITLE
Improved rclcpp docblock

### PR DESCRIPTION
**Pull request by ahcorde**
:factory_worker: 

Improved rclcpp docblock:

 - returns
 - params
 - throws
 - descriptions

Signed-off-by: ahcorde <ahcorde@gmail.com>

### CODE REVIEW

File: rclcpp/include/rclcpp/utilities.hpp [hunk 1]
```diff
@@ -136,7 +136,7 @@ remove_ros_arguments(int argc, char const * const argv[]);
  * If nullptr is given for the context, then the global context is used, i.e.
  * the context initialized by rclcpp::init().
  *
- * \param[in] context Check for shutdown of this Context.
+ * \param[in] context Optional Check for shutdown of this Context.
```
Comment (brawner) [inline rclcpp/include/rclcpp/utilities.hpp:139]:
```suggestion
 * \param[in] context Optional check for shutdown of this Context.
```

File: rclcpp/include/rclcpp/utilities.hpp [hunk 2]
```diff
@@ -150,7 +150,7 @@ ok(rclcpp::Context::SharedPtr context = nullptr);
  *
  * Deprecated, as it is no longer different from rcl_ok().
  *
- * \param[in] context Check for initialization of this Context.
+ * \param[in] context Optional Check for initialization of this Context.
```
Comment (brawner) [inline rclcpp/include/rclcpp/utilities.hpp:153]:
```suggestion
 * \param[in] context Optional check for initialization of this Context.
```

**Comment by brawner** [comment]
Looks like this might need to be rebased

File: rclcpp/include/rclcpp/node.hpp [hunk 3]
```diff
@@ -864,18 +869,40 @@ class Node : public std::enable_shared_from_this<Node>
...
   get_topic_names_and_types() const;
 
+  /// Return a map of existing service names to list of service types.
+  /**
+   * \return a map of existing service names to list of service types.
+   * \throws std::runtime_error anything that rcl_error can throw
+   */
   RCLCPP_PUBLIC
   std::map<std::string, std::vector<std::string>>
   get_service_names_and_types() const;
```
Comment (wjwwood) [inline rclcpp/include/rclcpp/node.hpp:888]: This doc string looks like a copy-paste error. There were some additional fixes for it, but the brief is still wrong.

Also, something odd is happening where the doc string for the next method, `count_publishers()`, is not rendering:

https://docs.ros2.org/foxy/api/rclcpp/classrclcpp_1_1Node.html#aa3df8a7d8b174f8049dfd2766f4b2eb9

Comment (wjwwood) [reply rclcpp/include/rclcpp/node.hpp:888]: https://github.com/ros2/rclcpp/pull/1719

**Pull request merged**